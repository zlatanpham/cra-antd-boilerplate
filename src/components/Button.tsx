import React from 'react';
import styled from 'styled-components';
import tw from 'twin.macro';

const StyledButton = styled.button`
  ${tw`bg-black text-white px-10 py-4 rounded-full hover:shadow-lg hover:bg-purple-600 hover:text-red-300 transition-all duration-200 hover:scale-110 transform`}
  font-size: 20px;
  font-weight: bold;
  border: none;
`;

export interface ButtonProps {
  className?: string;
}

const Button: React.FunctionComponent<ButtonProps> = ({
  children,
  className = '',
  ...rest
}) => (
  <StyledButton className={className} {...rest} data-testid="my-button">
    {children}
  </StyledButton>
);

export default Button;
