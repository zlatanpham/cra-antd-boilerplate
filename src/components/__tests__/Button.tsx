import React from 'react';
import { render } from '@testing-library/react';
import Button from '../Button';

describe('<Button/>', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Button>My Button</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should allow to pass custom CSS classes', () => {
    const { getByTestId } = render(
      <Button className="addition classes here">My Button</Button>,
    );
    const button = getByTestId('my-button');
    expect(button.getAttribute('class')).toContain('addition classes here');
  });
});
