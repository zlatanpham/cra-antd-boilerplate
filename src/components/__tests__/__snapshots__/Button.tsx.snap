// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Should match snapshot 1`] = `
.c0 {
  background-color: #000;
  color: #fff;
  padding-left: 2.5rem;
  padding-right: 2.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  border-radius: 9999px;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  --transform-translate-x: 0;
  --transform-translate-y: 0;
  --transform-rotate: 0;
  --transform-skew-x: 0;
  --transform-skew-y: 0;
  --transform-scale-x: 1;
  --transform-scale-y: 1;
  -webkit-transform: translateX(var(--transform-translate-x)) translateY(var(--transform-translate-y)) rotate(var(--transform-rotate)) skewX(var(--transform-skew-x)) skewY(var(--transform-skew-y)) scaleX(var(--transform-scale-x)) scaleY(var(--transform-scale-y));
  -ms-transform: translateX(var(--transform-translate-x)) translateY(var(--transform-translate-y)) rotate(var(--transform-rotate)) skewX(var(--transform-skew-x)) skewY(var(--transform-skew-y)) scaleX(var(--transform-scale-x)) scaleY(var(--transform-scale-y));
  transform: translateX(var(--transform-translate-x)) translateY(var(--transform-translate-y)) rotate(var(--transform-rotate)) skewX(var(--transform-skew-x)) skewY(var(--transform-skew-y)) scaleX(var(--transform-scale-x)) scaleY(var(--transform-scale-y));
  font-size: 20px;
  font-weight: bold;
  border: none;
}

.c0:hover {
  box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1),0 4px 6px -2px rgba(0,0,0,0.05);
  background-color: #805ad5;
  color: #feb2b2;
  --transform-scale-x: 1.1;
  --transform-scale-y: 1.1;
}

<button
  class="c0"
  data-testid="my-button"
>
  My Button
</button>
`;
